plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/CekMachineCosts.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE DataKinds #-}
+
+module UntypedPlutusCore.Evaluation.Machine.Cek.CekMachineCosts
+    ( CekMachineCosts(..)
+    , unitCekMachineCosts
+    )
+where
+
+import           PlutusCore.Evaluation.Machine.ExBudget
+
+import           Deriving.Aeson
+import           Language.Haskell.TH.Syntax             (Lift)
+
+-- | Costs for evaluating AST nodes.  Times should be specified in picoseconds, memory sizes in bytes.
+data CekMachineCosts =
+    CekMachineCosts {
+      cekStartupCost :: ExBudget  -- General overhead
+    , cekVarCost     :: ExBudget
+    , cekConstCost   :: ExBudget
+    , cekLamCost     :: ExBudget
+    , cekDelayCost   :: ExBudget
+    , cekForceCost   :: ExBudget
+    , cekApplyCost   :: ExBudget
+    , cekBuiltinCost :: ExBudget
+    -- ^ Just the cost of evaluating a Builtin node, not the builtin itself.
+    -- There's no entry for Error since we'll be exiting anyway; also, what would
+    -- happen if calling 'Error' caused the budget to be exceeded?
+    }
+    deriving (Show, Generic, Lift)
+    deriving (FromJSON, ToJSON) via CustomJSON '[FieldLabelModifier (CamelToSnake)] CekMachineCosts
+
+-- Charge a unit CPU cost for AST nodes: this allows us to count the number of
+-- times each node type is evaluated.  For actual prediction/costing we use
+-- a different version of CekMachineCosts: see ExBudgetingDefaults.defaultCekMachineCosts.
+unitCekMachineCosts :: CekMachineCosts
+unitCekMachineCosts =
+    CekMachineCosts zeroCost unitCost
```

Comment:
Nit: I'd make it

```haskell
CekMachineCosts
    { cekStartupCost = zeroCost
    , cekVarCost     = unitCost
    <...>
    }
```

Reads better and is immune to reordering of fields in `CekMachineCosts`.

Reply:
Right, that's a bit clearer.  I was just too lazy to type out all the field names.  I think when I wrote the first version of the everything was `unitCost` so it didn't matter too much.

Reply:
Right, that's a bit clearer.  I was just too lazy to type out all the field names.  I think when I wrote the first version of the everything was `unitCost` so it didn't matter too much.